src/Map/ProspectInfo.cs

```diff
@@ -167,6 +171,84 @@ private void ParseMessage(string message)
                 _foundOres.Add(val.Name);
         }
 
+        /// <summary>
+        /// Parses the legacy style messages (From VsProspectorInfo versions < 4.0.0).
+        /// Should only be called on load since it is significantly slower than <see cref="ParseMessage(string)"/>.
+        /// Checks every language to parse <see cref="Message"/> and sets the field to null afterwards.
+        /// If no language can be matched to the Message, nothing happens.
+        /// </summary>
+        private void ParseLegacyMessage()
```

Comment:
The parsing logic could be extracted into a separate helper class that accepts a ParseMessage with an optional language key (which default to the current one). That way the code does not have to be duplicated.
The class could cache a dictionary of regexes for each language key which gets populated on first parse message (or maybe for the default language at construction).

Reply:
I don't think that's necessary. ParseLegacyMessage is only called the first time an old safe is loaded. Caching it would maybe save up to a second of load time on large worlds. I don't think that's worth the work.

Also, there is no real code duplication as ParseMessage and ParseLegacyMessage work completely differently.
ParseMessage uses Regex to extract the information
while ParseLegacyMessage checks if the message contains any of the ore or density strings by comparing.